File: app/src/types/company-experience.ts
export interface CompanyExperienceResult {
  company?: string
  position?: string
  description?: string
  startDate?: string
  endDate?: string
  skills?: string[]
  achievements?: string[]
}


export interface CompanyExperienceForm {
  company?: string
  position?: string
  description?: string
  startDate?: string
  endDate?: string
  skills?: string[]
  achievements?: string[]
}

export interface CompanyExperienceQuery extends BodyQuery {

}


export interface CompanyExperienceItem {

}

File: server/src/main/java/cn/net/yunlou/bole/model/entity/CompanyExperience.java
package cn.net.yunlou.bole.model.entity;

import cn.net.yunlou.bole.common.BaseEntity;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonFormat;
import java.time.LocalDate;
import lombok.*;

@EqualsAndHashCode(callSuper = true)
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName(value = "t_company_experiences")
public class CompanyExperience extends BaseEntity {

    private Long userId;

    private Long companyId;

    private String position;

    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate startDate;

    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate endDate;

    private Boolean isCurrent;

    private String description;

    private Integer sort;

    @TableField(exist = false)
    private Company company;
}
